source/MaterialXFormat/XmlIo.cpp

```diff
@@ -55,7 +55,14 @@ void elementFromXml(const xml_node& xmlNode, ElementPtr elem, const XmlReadOptio
         ConstElementPtr previous = elem->getChild(name);
         if (previous)
         {
-            continue;
+            if (!readOptions || !readOptions->generateUniqueNames)
```

Comment:
New read option for generatingUniqueNames.

Reply:
Looks good. Thanks.